crates/symbolicator/src/services/symcaches.rs

```diff
@@ -369,10 +369,163 @@ fn write_symcache(
+        test::setup();
+
+        const TIMEOUT: Duration = Duration::from_millis(500);
+
+        let cache_dir = test::tempdir();
+        let symbol_dir = test::tempdir();
+
+        // Create directories for the symbol file and the bcsymbolmap
+        let macho_dir = symbol_dir.path().join("2d/10c42f591d3265b14778ba0868073f/");
+        let symbol_map_dir = symbol_dir.path().join("c8/374b6d6e9634d8ae38efaa5fec424f/");
+
+        fs::create_dir_all(&symbol_map_dir).unwrap();
+        fs::create_dir_all(&macho_dir).unwrap();
+
+        // Copy the symbol file to the temporary symbol directory
+        fs::copy(
+            fixture("symbols/2d10c42f-591d-3265-b147-78ba0868073f.dwarf-hidden"),
+            macho_dir.join("debuginfo"),
+        )
+        .unwrap();
+
+        let source = SourceConfig::Filesystem(Arc::new(FilesystemSourceConfig {
+            id: SourceId::new("local"),
+            path: symbol_dir.path().to_owned(),
+            files: CommonSourceConfig::with_layout(DirectoryLayoutType::Unified),
+        }));
+
+        let identifier = ObjectId::from(DebugId::from_uuid(
+            Uuid::parse_str("2d10c42f-591d-3265-b147-78ba0868073f").unwrap(),
+        ));
+
+        let fetch_symcache = FetchSymCache {
+            object_type: ObjectType::Macho,
+            identifier,
+            sources: Arc::new([source]),
+            scope: Scope::Global,
+        };
+
+        let symcache_actor = symcache_actor(cache_dir.path().to_owned(), TIMEOUT);
+
+        test::spawn_compat(move || async move {
+            // Create the symcache for the first time. Since the bcsymbolmap is not available, names in the
+            // symcache will be obfuscated.
+            let symcache_file = symcache_actor.fetch(fetch_symcache.clone()).await.unwrap();
+            let symcache = symcache_file.parse().unwrap().unwrap();
+            let line_info = symcache.lookup(0x5a75).unwrap().next().unwrap().unwrap();
+            assert_eq!(line_info.filename(), "__hidden#42_");
+            assert_eq!(line_info.symbol(), "__hidden#0_");
+
+            // Copy the plist and bcsymbolmap to the temporary symbol directory so that the SymCacheActor can find them.
+            fs::copy(
+                fixture("symbols/2d10c42f-591d-3265-b147-78ba0868073f.plist"),
+                macho_dir.join("uuidmap"),
+            )
+            .unwrap();
+
+            fs::copy(
+                fixture("symbols/c8374b6d-6e96-34d8-ae38-efaa5fec424f.bcsymbolmap"),
+                symbol_map_dir.join("bcsymbolmap"),
+            )
+            .unwrap();
+
+            // Create the symcache for the second time. Even though the bcsymbolmap is now available, its absence should
+            // still be cached and the SymcacheActor should make no attempt to download it. Therefore, the names should
+            // be obfuscated like before.
+            let symcache_file = symcache_actor.fetch(fetch_symcache.clone()).await.unwrap();
```

Comment:
I wonder if we also get the `CacheStatus` in the result to assert that?

Reply:
I think we get the cache status of the symcache file, but is that relevant?